Sources/Receipt/ReceiptDetailTableViewController.swift

```diff
@@ -0,0 +1,119 @@
+    ]
+
+    private var presenter: ReceiptDetailViewPresenter!
+
+    public init(with hyperwalletReceipt: HyperwalletReceipt) {
+        super.init(nibName: nil, bundle: nil)
+        presenter = ReceiptDetailViewPresenter(with: hyperwalletReceipt)
+    }
+
+    // swiftlint:disable unavailable_function
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("NSCoding not supported")
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        title = "title_receipts_details".localized()
+        titleDisplayMode(.never)
+        setViewBackgroundColor()
+        setupReceiptDetailTableView()
+    }
+
+    private func setupReceiptDetailTableView() {
+        tableView = UITableView(frame: view.frame, style: .grouped)
+        tableView.allowsSelection = false
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.estimatedRowHeight = UITableView.automaticDimension
+        tableView.accessibilityIdentifier = "receiptDetailTableView"
+        registeredCells.forEach {
+            tableView.register($0.type, forCellReuseIdentifier: $0.id)
+        }
+    }
+
+    private func cellForRowAt(_ indexPath: IndexPath) -> UITableViewCell {
+        let cellIdentifier = presenter.sectionData[indexPath.section].cellIdentifier
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)
+        cell.separatorInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: .greatestFiniteMagnitude)
+        let section = presenter.sectionData[indexPath.section]
+        switch section.receiptDetailSectionHeader {
+        case .transaction:
+            if let tableViewCell = cell as? ReceiptTransactionTableViewCell,
+                let transactionSection = section as? ReceiptDetailSectionTransactionData {
+                tableViewCell.configure(configuration: transactionSection.tableViewCellConfiguration)
+            }
+
+        case .details:
+            if let tableViewCell = cell as? ReceiptDetailTableViewCell,
+                let detailSection = section as? ReceiptDetailSectionDetailData {
+                let row = detailSection.rows[indexPath.row]
+                tableViewCell.textLabel?.text = row.title
+                tableViewCell.detailTextLabel?.text = row.value
+            }
+
+        case .fee:
+            if let tableViewCell = cell as? ReceiptFeeTableViewCell,
+                let feeSection = section as? ReceiptDetailSectionFeeData {
+                let row = feeSection.rows[indexPath.row]
+                tableViewCell.textLabel?.text = row.title
+                tableViewCell.detailTextLabel?.text = row.value
+            }
+
+        case .notes:
+            if let tableViewCell = cell as? ReceiptNotesTableViewCell,
+                let feeSection = section as? ReceiptDetailSectionNotesData {
+                tableViewCell.textLabel?.text = feeSection.notes!
```

Comment:
We should not force unwrap notes because the note section should not display at all.

Reply:
fixed